addons/pos_loyalty/models/loyalty_reward.py

```diff
@@ -11,3 +11,8 @@ def _get_discount_product_values(self):
         for vals in res:
             vals.update({'taxes_id': False})
         return res
+
+    def unlink(self):
+        if self.env['pos.order.line'].search_count([('reward_id', 'in', self.ids)], limit=1):
+            return self.action_archive()
```

Comment:
```suggestion
        if len(self) == 1 and self.env['pos.order.line'].search_count([('reward_id', 'in', self.ids)], limit=1):
            return self.action_archive()
```
You shouldn't archive all rewards if only one cannot be deleted.
I think that focusing on the cases where the users deletes one reward at a time should be enough for now, since that's how it works from the interface.